File: backend/src/projects/services/project.service.ts
import { Injectable } from '@nestjs/common';
import { PrismaService } from '../../prisma/prisma.service';
import { IProjectService } from '../interfaces/project.service.interface';
import { CreateProjectDto, ProjectDto } from '../dto/project.dto';

/**
 * Сервис для управления проектами
 */
@Injectable()
export class ProjectService implements IProjectService {
  constructor(private readonly prisma: PrismaService) {}

  /**
   * Создает новый проект
   * @param createProjectDto Данные для создания проекта
   * @returns Созданный проект
   */
  async createProject(createProjectDto: CreateProjectDto): Promise<ProjectDto> {
    const project = await this.prisma.project.create({
      data: {
        name: createProjectDto.name,
      },
    });

    return {
      id: project.id,
      name: project.name,
      createdAt: project.createdAt,
      updatedAt: project.updatedAt,
    };
  }

  /**
   * Получает проект по идентификатору
   * @param id Идентификатор проекта
   * @returns Проект или null, если не найден
   */
  async getProjectById(id: string): Promise<ProjectDto | null> {
    const project = await this.prisma.project.findUnique({
      where: { id },
    });

    if (!project) {
      return null;
    }

    return {
      id: project.id,
      name: project.name,
      createdAt: project.createdAt,
      updatedAt: project.updatedAt,
    };
  }

  /**
   * Получает все проекты
   * @returns Список всех проектов
   */
  async getAllProjects(): Promise<ProjectDto[]> {
    const projects = await this.prisma.project.findMany({
      orderBy: { createdAt: 'desc' },
    });

    return projects.map(project => ({
      id: project.id,
      name: project.name,
      createdAt: project.createdAt,
      updatedAt: project.updatedAt,
    }));
  }
}

File: backend/src/projects/services/ProjectService.cs
using System;
using System.Linq;
using System.Threading.Tasks;
using Microsoft.EntityFrameworkCore;
using Testoria.Projects.DTO;
using Testoria.Projects.Exceptions;
using Testoria.Projects.Interfaces;

namespace Testoria.Projects.Services
{
    /// <summary>
    /// Сервис для управления проектами
    /// </summary>
    public sealed class ProjectService : IProjectService
    {
        private readonly PrismaService _prismaService;

        /// <summary>
        /// Инициализирует новый экземпляр класса ProjectService
        /// </summary>
        /// <param name="prismaService">Сервис для работы с базой данных</param>
        public ProjectService(PrismaService prismaService)
        {
            _prismaService = prismaService;
        }

        /// <inheritdoc />
        public async Task<ProjectDto> CreateProjectAsync(CreateProjectDto createProjectDto)
        {
            var project = await _prismaService.Project.CreateAsync(new()
            {
                Name = createProjectDto.Name
            });

            return new ProjectDto
            {
                Id = Guid.Parse(project.Id),
                Name = project.Name,
                CreatedAt = project.CreatedAt,
                UpdatedAt = project.UpdatedAt
            };
        }

        /// <inheritdoc />
        public async Task<ProjectDto?> GetProjectByIdAsync(Guid id)
        {
            var project = await _prismaService.Project.FindUniqueAsync(p => p.Id == id.ToString());

            if (project == null)
            {
                return null;
            }

            return new ProjectDto
            {
                Id = Guid.Parse(project.Id),
                Name = project.Name,
                CreatedAt = project.CreatedAt,
                UpdatedAt = project.UpdatedAt
            };
        }

        /// <inheritdoc />
        public async Task<ProjectDto[]> GetAllProjectsAsync()
        {
            var projects = await _prismaService.Project.FindManyAsync();

            return projects.Select(p => new ProjectDto
            {
                Id = Guid.Parse(p.Id),
                Name = p.Name,
                CreatedAt = p.CreatedAt,
                UpdatedAt = p.UpdatedAt
            }).ToArray();
        }
    }
}
